src/raft_server.c

```diff
@@ -192,9 +179,20 @@ int raft_become_leader(raft_server_t* me_)
 
         raft_node_set_next_idx(node, next_idx);
         raft_node_set_match_idx(node, 0);
-        raft_send_appendentries(me_, node);
     }
 
+    raft_entry_t *noop = raft_entry_new(0);
+    noop->term = raft_get_current_term(me_);
+    noop->type = RAFT_LOGTYPE_NO_OP;
+    int e = raft_recv_entry(me_, noop, &response);
+    if (0 != e)
+        return e;
+    raft_entry_release(noop);
```

Comment:
```suggestion

    int e = raft_recv_entry(me_, noop, &response);
    raft_entry_release(noop);
    
    if (0 != e)
        return e;
```

Let's fix the memory leak in the way.

Reply:
Ahh, also we may want to revert back to follower if we get an error here? (I know we are not doing good with error handling in the library right now but let's match the previous behavior, it was setting itself "leader" after the append. If we go back to follower on error, we'll match the previous behavior I believe)